snippets/puppetlabs_repo.erb

```diff
@@ -0,0 +1,23 @@
+<%#
+kind: snippet
+name: puppetlabs_repo
+-%>
+<%
+  http_proxy = @host.params['http-proxy'] ? " --httpproxy #{@host.params['http-proxy']}" : nil
+  http_port = @host.params['http-proxy-port'] ? " --httpport #{@host.params['http-proxy-port']}" : nil
+  os_major = @host.operatingsystem.major.to_i
+-%>
+<% if @host.param_true?('enable-puppetlabs-repo') -%>
+<% if @host.operatingsystem.name == 'Fedora' -%>
+rpm -Uvh<%= http_proxy %><%= http_port %> <%= @host.os.medium_uri(@host, "http://yum.puppetlabs.com/puppetlabs-release-fedora-#{os_major}.noarch.rpm") %>
+<% else -%>
```

Comment:
We'll need support for Debian/Ubuntu in here too I think, or at the _very_ least a check that doesn't try to run `rpm -Uvh` on a non-EL box.